syntax = "proto3";

package yandex.cloud.loadtesting.agent.v1;

import "operation.proto";
import "test_service.proto";

option go_package = "agent";


service AgentRegistrationService {

  rpc Register (RegisterRequest) returns (RegisterResponse);
  rpc ExternalAgentRegister (ExternalAgentRegisterRequest) returns (operation.Operation) {
    option (operation) = {
      metadata: "ExternalAgentRegisterMetadata"
      response: "AgentInstance"
    };
  };

}

message RegisterRequest {
  string compute_instance_id = 1;
}

message RegisterResponse {
  string agent_instance_id = 1;
}

message ExternalAgentRegisterRequest {
  string folder_id = 1;
  string compute_instance_id = 2;
  string name = 3;
  string agent_version = 4;
}

message ExternalAgentRegisterMetadata {
  string agent_instance_id = 1;
}
